src/platforms/posix/drivers/df_ms5611_wrapper/df_ms5611_wrapper.cpp

```diff
@@ -0,0 +1,322 @@
+
+
+class DfMS5611Wrapper : public MS5611
+{
+public:
+	DfMS5611Wrapper();
+	~DfMS5611Wrapper();
+
+
+	/**
+	 * Start automatic measurement.
+	 *
+	 * @return 0 on success
+	 */
+	int		start();
+
+	/**
+	 * Stop automatic measurement.
+	 *
+	 * @return 0 on success
+	 */
+	int		stop();
+
+private:
+	int _publish(struct baro_sensor_data &data);
+
+	orb_advert_t		_baro_topic;
+
+	int			_baro_orb_class_instance;
+
+	perf_counter_t		_baro_sample_perf;
+
+};
+
+DfMS5611Wrapper::DfMS5611Wrapper() :
+	MS5611(BARO_DEVICE_PATH),
+	_baro_topic(nullptr),
+	_baro_orb_class_instance(-1),
+	_baro_sample_perf(perf_alloc(PC_ELAPSED, "df_baro_read"))
+{
+}
+
+DfMS5611Wrapper::~DfMS5611Wrapper()
+{
+	perf_free(_baro_sample_perf);
+}
+
+int DfMS5611Wrapper::start()
+{
+	/* Init device and start sensor. */
+	int ret = init();
+
+	if (ret != 0) {
+		PX4_ERR("MS5611 init fail: %d", ret);
+		return ret;
+	}
+
+	ret = MS5611::start();
+
+	if (ret != 0) {
+		PX4_ERR("MS5611 start fail: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+int DfMS5611Wrapper::stop()
+{
+	/* Stop sensor. */
+	int ret = MS5611::stop();
+
+	if (ret != 0) {
+		PX4_ERR("MS5611 stop fail: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+int DfMS5611Wrapper::_publish(struct baro_sensor_data &data)
+{
+	perf_begin(_baro_sample_perf);
+
+	baro_report baro_report = {};
+	baro_report.timestamp = hrt_absolute_time();
+
+	baro_report.pressure = data.pressure_pa;
+	baro_report.temperature = data.temperature_c;
+
+	// TODO: verify this, it's just copied from the MS5611 driver.
+
+	// Constant for now
+	const double MSL_PRESSURE_KPA = 101325.0 / 1000.0;
+
+	/* tropospheric properties (0-11km) for standard atmosphere */
+	const double T1 = 15.0 + 273.15;	/* temperature at base height in Kelvin */
+	const double a  = -6.5 / 1000;	/* temperature gradient in degrees per metre */
+	const double g  = 9.80665;	/* gravity constant in m/s/s */
+	const double R  = 287.05;	/* ideal gas constant in J/kg/K */
+
+	/* current pressure at MSL in kPa */
+	double p1 = MSL_PRESSURE_KPA;
+
+	/* measured pressure in kPa */
+	double p = static_cast<double>(data.pressure_pa) / 1000.0;
+
+	/*
+	 * Solve:
+	 *
+	 *     /        -(aR / g)     \
+	 *    | (p / p1)          . T1 | - T1
+	 *     \                      /
+	 * h = -------------------------------  + h1
+	 *                   a
+	 */
+	baro_report.altitude = (((pow((p / p1), (-(a * R) / g))) * T1) - T1) / a;
+
+	// TODO: when is this ever blocked?
+	if (!(m_pub_blocked)) {
+
+		if (_baro_topic == nullptr) {
+			_baro_topic = orb_advertise_multi(ORB_ID(sensor_baro), &baro_report,
+							  &_baro_orb_class_instance, ORB_PRIO_DEFAULT);
+
+		} else {
+			orb_publish(ORB_ID(sensor_baro), _baro_topic, &baro_report);
+		}
+	}
+
+	/* Notify anyone waiting for data. */
+	DevMgr::updateNotify(*this);
+
+	perf_end(_baro_sample_perf);
+
+	return 0;
+};
+
+
+namespace df_ms5611_wrapper
+{
+
+DfMS5611Wrapper *g_dev = nullptr;
+
+int start(/* enum Rotation rotation */);
```

Comment:
You might be able to delete the rotation stuff because that's not applicable to baros :).